### PR TITLE
docs: Add Docserv² AsciiDoc docs

### PR DESCRIPTION
**Pull request by sknorr**
@tomschr Hi. Could please review these configuration docs?

For your reading pläscha, also uploaded to https://opensuse.github.io/docserv/

### CODE REVIEW

File: docs/adoc/config-site.adoc [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+* A set of language codes that navigational pages are created for (`languages` attribute, target-specific)
+====
+
+
+[#config-site-target-section]
+## Target sections
+
+{ds2} supports reuse of the same `*docserv*` process to synchronize to multiple target servers.
+Each of those target servers is defined in its own target section.
+All or a subset of target servers can share certain parts of their configuration by sharing the specific configuration directories, if desired.
```
Comment (tomschr) [inline docs/adoc/config-site.adoc:100]: I would also mention that some keys allows *relative paths*. In the list below I would mention which keys allows this.

File: docs/adoc/config-site.adoc [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+site_sections = section-a section-b
+default_site_section = section-a
+
+zip_formats = pdf epub single-html
+----
+
+`[target+++_+++__n__]`::
+  The section identifier which needs to start with `target_`, followed by an integer.
+
+`name`::
```
Comment (tomschr) [inline docs/adoc/config-site.adoc:142]: Not sure if it make sense, but having the allowed datatype could be helpful.

**Review by tomschr — CHANGES_REQUESTED** [review]
@sknorr Thanks for writing down all of docserv². Much appreciated! :heart: 
I found some smaller issues, nothing serious. Hope you find them useful.

File: config/docserv.ini [hunk 3]
```diff
@@ -69,8 +69,7 @@ fragment_dir = /etc/docserv/fragment-internal
 fragment_l10n_dir = /etc/docserv/fragment-l10n-internal
 # Sections of the published site. Each section gets an own top-level
 # navigation page. XML product configuration files can be assigned to a site
-# section using the <product site-section="..."/> attribute. Each site section
-# needs an HTML template file named template-[SITE_SECTION].html.
+# section using the `<product site-section="..."/>` attribute.
 site_sections = main best-practices
```
[the remark this answers is not in the document]
Comment (sknorr) [reply config/docserv.ini:73]: No

File: docs/adoc/0-docserv-guide.adoc [hunk 4]
```diff
@@ -0,0 +1,23 @@
+include::1-attribute.adoc[]
+
+[#docserv2-guide]
+# {ds2} Guide
```
[the remark this answers is not in the document]
Comment (sknorr) [reply docs/adoc/0-docserv-guide.adoc:4]: That's not what it is though. It tells you how to set up a site with Ds2.

File: docs/adoc/config-localization.adoc [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+There must be at least one localization file for the default UI language.
+This file must contain all requested localization strings.
+
+Files for alternative UI languages and the strings within them are optional.
+In case they are missing, the default UI language's string will be used as a fallback.
+
+[#config-localization-file]
+## Localization file structure
+
+Localization files are based on an easily validatable XML schema.
```
[the remark this answers is not in the document]
Comment (sknorr) [reply docs/adoc/config-localization.adoc:20]: So, I was not 100% sure about this wording either. The idea here is that this format is intentionally simple and easy to validate. I would not focus on the RNG v/ XSD thing. Suggestion:

```suggestion
Localization files are based on a simple file format and easy to validate.
```

File: docs/adoc/config-localization.adoc [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+  <content key="LOCALIZATION_KEY">LOCALIZED_CONTENT</content>
+
+</locale>
+----
+
+* The `__LANGUAGE_CODE__` must be of the type `la-ng` (for example, `en-us`).
+This language code is purely informative.
+{ds2} will only pick up the language code from the file name.
+
+* The `__LOCALIZATION_KEY__` must be unique within each localization file.
```
[the remark this answers is not in the document]
Comment (sknorr) [reply docs/adoc/config-localization.adoc:43]: It's an xsd:ID type.

File: docs/adoc/config-product.adoc [hunk 7]
```diff
@@ -0,0 +1,306 @@
...
+
+* DC files for documents in the primary language are kept in the root directory of the repository.
+
+* A DC file named with its name ending in -all references a set of books and articles.
+
+Its usefulness degrades the fewer of these conventions are implemented in the Git source repository.
+
+To add a new product, proceed as follows:
+
+. Install the RPM package docserv from the external Documentation:Tools OBS repository.
```
[the remark this answers is not in the document]
Comment (sknorr) [reply docs/adoc/config-product.adoc:268]: Ugh ... That sentence should not be there at all, doesn't make sense as part of the topic. I will need to fix that.

File: docs/adoc/config-site.adoc (hunk 2 above)
Comment (sknorr) [reply docs/adoc/config-site.adoc:142]: I guess the issue is that we'd need to properly define data types then. :)

File: docs/adoc/config-site.adoc [hunk 8]
```diff
@@ -0,0 +1,278 @@
...
+`repo_dir`::
+  Specifies the directory that is used to permanently cache remote Git repositories.
+  Git repositories in this directory persist across builds and instances of the {ds2} process.
+
+`temp_repo_dir`::
+  Specifies the directory that is used to temporarily cache specific Git branches.
+  When receiving a build instruction, {ds2} will clone an individual branch of a Git repository from `repo_dir`.
+  It will then build using the data from `temp_repo_dir`.
+  This prevents branch checkout collisions during builds and other problems.
+  After the build instruction has run through,
```
[the remark this answers is not in the document]
Comment (sknorr) [reply docs/adoc/config-site.adoc]: ... fixed!

File: docs/adoc/config-site.adoc (hunk 1 above)
Comment (sknorr) [reply docs/adoc/config-site.adoc:104]: Done.

File: config/docserv.ini (hunk 3 above)
[the remark this answers is not in the document]
Comment (sknorr) [reply config/docserv.ini:73]: Ahhh... Wait, I misunderstood your question.
Yes, the order of site sections is relevant: The section listed first gets the first tab.

[the remark this answers is not in the document]
Comment (tomschr) [reply config/docserv.ini:73]: Thanks for the clarification. :+1:

File: docs/adoc/config-localization.adoc (hunk 5 above)
[the remark this answers is not in the document]
Comment (tomschr) [reply docs/adoc/config-localization.adoc:20]: That would work too. 

The schema file is `share/build-navigation/fragment-l10n-schema.rnc`, right?

File: docs/adoc/config-site.adoc (hunk 2 above)
Comment (tomschr) [reply docs/adoc/config-site.adoc:146]: I wouldn't expect a rigid or strict datatype system. I would be happy to have some basic types like string, bool, integer etc. Maybe even a URL type to distinguish from string. Just to get an idea what is expected.

File: docs/adoc/config-localization.adoc (hunk 5 above)
[the remark this answers is not in the document]
Comment (sknorr) [reply docs/adoc/config-localization.adoc:20]: It is.

File: docs/adoc/config-product.adoc [hunk 9]
```diff
@@ -0,0 +1,325 @@
...
+* A reference to the default docset of a product: `<ref product="__PRODUCT_ID__"/>`
+
+* A reference to a specific docset of a product: `<ref product="__PRODUCT_ID__" docset="__SET_ID__"/>`
+
+* A reference to a specific DC definition within `<builddocs/>`: `<ref product="__PRODUCT_ID__" docset="__SET_ID__" dc="__DC_NAME__"/>`
+
+* A reference to a specific DC definition that has subdeliverables within `<builddocs/>`: `<ref product="__PRODUCT_ID__" docset="__SET_ID__" dc="__DC_NAME__" subdeliverable="__SUBDELIVERABLE_NAME__"/>`
+
+* A reference to a specific external link: `<ref product="__PRODUCT_ID__" docset="__SET_ID__" link="__LINK_ID__"/>`
+The value for `__LINK_ID__` has to be specifically set on the corresponding `<link/>` element by adding the attribute `<link linkid="__NAME_OF_ID__"/>`. `__NAME_OF_ID__ must be unique within the current `<docset/>` and may consist of talphanumeric characters and the characters `_-.`.
```
Comment (quatran) [inline docs/adoc/config-product.adoc:219]: Typo.

```suggestion
The value for `__LINK_ID__` has to be specifically set on the corresponding `<link/>` element by adding the attribute `<link linkid="__NAME_OF_ID__"/>`. `__NAME_OF_ID__ must be unique within the current `<docset/>` and may consist of alphanumeric characters and the characters `_-.`.
```

Comment (sknorr) [reply docs/adoc/config-product.adoc:219]: Thanks! I resolved this locally after you told me via chat before I saw this comment. Resolving.

File: docs/adoc/config-localization.adoc (hunk 6 above)
[the remark this answers is not in the document]
Comment (sknorr) [reply docs/adoc/config-localization.adoc]: Noted in the docs now, resolving.

File: docs/adoc/config-site.adoc (hunk 2 above)
Comment (sknorr) [reply docs/adoc/config-site.adoc]: I have added such information, resolving this too.

**Comment by sknorr** [comment]
@tomschr Mind going over this another time? I have added a bunch of new content too (installing, backups, API) and I think we should be suitably close to this being mergable. URL is still https://opensuse.github.io/docserv/.

File: docs/adoc/intro-overview.adoc [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+
+image::docserv2-overview.svg[title="{ds2} overview",width=90%]
+
+The following elements that are relevant to {ds2}:
+
+* Machines:
+
+** Git is used for versioning the sources of the content to publish.
+
+** The configuration for {ds2} should be stored in a Git repository as well, though a separate one to that of the documentation.
```
Comment (tomschr) [inline docs/adoc/intro-overview.adoc:21]: Git and configuration files grouped under "Machines"? Hmn, I don't get that. Actually, it confused me. Maybe find a different category for these two points?

What about "Version control config files"? Or, if you prefer it more general, "Basics"?

File: docs/adoc/intro-terminology.adoc [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+
+// We would have loved to use a variablelist::, but the default DocBook
+// stylesheets do not support using xrefs to <term/>s (or rather, don't generate
+// xref titles in that case).
+
+
+[#term-build-instruction]
+## Build instruction
+
+An instruction to build (or rebuild) a specific set of documents in specific language.
```
Comment (tomschr) [inline docs/adoc/intro-terminology.adoc:16]: Missing an "a"?

```suggestion
An instruction to build (or rebuild) a specific set of documents in a specific language.
```

**Review by tomschr — CHANGES_REQUESTED** [review]
@sknorr Thanks for your efforts! Great work! :+1:

I looked through it. Found some minor things and added some questions. Some are a matter of taste. Hope it helps.

File: docs/adoc/api.adoc [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+> curl \
+    --header "Content-Type: application/json" \
+    --request POST \
+    --data '{"target":"__[TARGET]__","product":"__[PRODUCT]__","docset":"__[DOCSET]__","lang":"__[LANGUAGE_CODE]__"}' \
+    __[HOST]__:__[PORT]__
+----
+
+Replace `__[TARGET]__`, `__[PRODUCT]__`, `__[DOCSET]__`, `__[LANGUAGE_CODE]__`, `__[HOST]__`, and `__[PORT]__` with appropriate values.
+You can combine multiple build instructions into a single `POST` requests by sending a data field with multiple JSON objects separated with comma characters (`,`).
+
```
[the remark this answers is not in the document]
Comment (sknorr) [reply docs/adoc/api.adoc:65]: I mean, there is an example right above...l.56 to 61.

File: docs/adoc/backup.adoc [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+If it is deleted, all metadata will have to be created anew.
+This means, all docsets of the site have to be rebuilt which can take a substantial amount of time.
+If you do not do so, navigational pages will lose links to most content after the immediate next build.
+
+To maintain and back up the metadata store, create a Git repository at `/var/cache/docserv/`.
+To do so, run `git -C /var/cache/docserv/ init`.
+Create a `.gitignore` file that ignores `.json` files on the top level of the repository to avoid backing up build queue files.
+Then add, commit, and push all content as desired.
+
+It usually makes sense to create a cronjob that automatically adds/commits/pushes all changes on a regular basis.
```
[the remark this answers is not in the document]
Comment (sknorr) [reply docs/adoc/backup.adoc:37]:
```suggestion
It usually makes sense to create a cronjob or similar that automatically adds/commits/pushes all changes on a regular basis.
```

File: docs/adoc/intro-overview.adoc [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+
+** The {ds2} build server is the core piece of the publication process.
+It builds documents and their associated navigation pages.
+After building a specific piece of documentation, it copies the documentation output to a backup directory and from there to a publication server.
```
Comment (sknorr) [inline docs/adoc/intro-overview.adoc:25]: [toms](https://github.com/openSUSE/docserv/pull/279/files#r782109023):
> Git and configuration files grouped under "Machines"? Hmn, I don't get that. Actually, it confused me. Maybe find a different category for these two points?

You're right ... That did not belong there.

```suggestion
** The {ds2} build server is the core piece of the publication process.
It builds documents and their associated navigation pages.
After building a specific piece of documentation, it copies the documentation output to a backup directory and from there to a publication server.
```

File: docs/adoc/intro-overview.adoc (hunk 10 above)
Comment (sknorr) [reply docs/adoc/intro-overview.adoc]: Removed the point.

**Pull request merged**